agent/proxycfg/snapshot.go

```diff
@@ -191,6 +245,8 @@ func (s *ConfigSnapshot) Valid() bool {
 	switch s.Kind {
 	case structs.ServiceKindConnectProxy:
 		return s.Roots != nil && s.ConnectProxy.Leaf != nil
+	case structs.ServiceKindTerminatingGateway:
+		return s.Roots != nil
```

Comment:
This implies a TG should start serving traffic if it _just_ has CA roots. Would you prefer to wait for the relevant leafs to come in as well? Should you also wait until you know about the addresses of the things you are terminating for?

Reply:
Well, it's not going to serve traffic until those leaves/endpoints are present, it's only going to allow connections for health checks to pass. The listeners/clusters/endpoints won't be created without addresses or leaves. 

I could do something like MeshGateways where they set a flag once those watches are known to be running (even if the response is empty):
https://github.com/hashicorp/consul/blob/a1648c61ae0aa5c07fa8f17fbc67dd1d8db619ff/agent/proxycfg/snapshot.go#L68-L74

Thoughts?

Reply:
After thinking about it some more, I don't see the need to avoid configuring Envoy if there are issues contacting the servers on the initial blocking query. 

If those issues are aren't transient, then they'll be visible in the agent's node health. Once they're resolved the watches should resume.